### PR TITLE
Upgrade Angular version to v12

### LINKED ISSUE #1365: [Housekeeping] Enforce uniform line endings
**Short description of housekeeping request**

Add `"endOfLine": "lf"` to `.prettierrc.json` to enforce Line Feed only line endings.

By enforcing a specific value for line endings we can avoid unwanted (potentially large) diffs and remove "noise" once and for all.

`lf` is the default value in Prettier since v2.0.0 and the most common inside git repos.

The Prettier docs mention that Windows users may need to re-clone the repo. Devs working on Kirby on Windows computers should be informed and advised to re-clone the repo.

**Alternatives**

`"endOfLine": "auto"` but may still cause unwanted diffs (at least until all files are normalised).

From Prettier docs:

> `"auto"`: Maintain existing line endings (mixed values within one file are normalised by looking at what’s used after the first line)

`auto` has the advantage that Windows devs don't need to re-clone the repo. However, every dev on the Kirby core team uses MacOS which suggests that we choose the `lf` option.

**Additional context**

https://prettier.io/docs/en/options.html#end-of-line has a step by step description for making sure the entire git repository only contains Linux-style line endings.

http://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration#Formatting-and-Whitespace

### Kick Off

* [ ] Ensure this issue has been `Tech refined` with @kirbydesign/kirby-guild and is updated with a clear implementation description
  _This issue should be in the [Ready to do](https://github.com/kirbydesign/designsystem/projects/1#column-4590936) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1) before starting implementation_
* [ ] Assign yourself to this issue and move it to the [In progress](https://github.com/kirbydesign/designsystem/projects/1#column-4590937) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1)

### Code

* [ ] Create Feature Branch from [master branch](https://github.com/kirbydesign/designsystem/tree/master)
* [ ] Implement/update unit tests
* [ ] Create a draft implementation and push to Github
* [ ] Ask a member of @kirbydesign/kirby-guild for a WIP review by creating a draft Pull Request

### Code Review

* [ ] Open a pull request (or mark the existing draft PR as `Ready for review`) and ask @kirbydesign/kirby-guild for a review
  _Remember to add `closes #issueno` to the description of the PR._
* [ ] Once approved, merge feature branch/PR to master

:tada: Celebrate

### LINKED ISSUE #1931: [Housekeeping] Upgrade to Angular 12
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the housekeeping request
We need to upgrade to Angular 12, to support the new Ionic 6 version.
See more here: https://ionicframework.com/docs/next/intro/upgrading-to-ionic-6

## Have you considered any alternatives?
No, we should definitely do this! 😄

### PR DESCRIPTION
**Pull request by RasmusKjeldgaard**
## Which issue does this PR close?

This PR closes #1931 closes #1365

## What is the new behavior?
We should now be up and running on v12. This changes is a first step in [upgrading to Ionic 6](https://ionicframework.com/docs/intro/upgrading-to-ionic-6). 

The diff is large, but that is solely due to the updated prettier version, and subsequently running `prettier:fix` to bring all files up to speed.

The somewhat interesting files should be: 
 - [package.json](https://github.com/kirbydesign/designsystem/blob/00c69ca42bd08e314735e42f8d6f2eae170efddc/package.json)
 - [annotations.config.ts](https://github.com/kirbydesign/designsystem/blob/00c69ca42bd08e314735e42f8d6f2eae170efddc/libs/designsystem/src/lib/components/chart/configs/annotations.config.ts)
 - [test-utils.ts](https://github.com/kirbydesign/designsystem/blob/00c69ca42bd08e314735e42f8d6f2eae170efddc/libs/designsystem/src/lib/components/chart/chart-js/test-utils.ts)
- [angular.json](https://github.com/kirbydesign/designsystem/blob/00c69ca42bd08e314735e42f8d6f2eae170efddc/angular.json)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Are there any additional context?
Prettier has been run on all the files in the project, as they were lagging behind. Lint-staged should take care of running it on all subsequent commits, via our precommit hook. 

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] ~~Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".~~
- [ ] ~~Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).~~

### Review  
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [ ] ~~Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/master/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)~~

When the pull request has been approved it will be automatically merged to master via [automerge](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/automatically-merging-a-pull-request).

### CODE REVIEW

File: apps/cookbook/src/app/component-status/component-status.component.scss [hunk 1]
```diff
@@ -126,7 +126,8 @@
     }
   }
 
-  a[href^="https://"] ion-icon {
+  a[href^="https://"] ion-icon
+  {
```
Comment (jkaltoft) [inline apps/cookbook/src/app/component-status/component-status.component.scss:130]: Do you have any idea why prettier did this?

Comment (RasmusKjeldgaard) [reply apps/cookbook/src/app/component-status/component-status.component.scss:130]: Not 100% sure about the root cause, but it has to do with the double forward slash. 
If I escape them, the style rule still works as expected, and prettier treats it like the rest of the code.

image

Comment (RasmusKjeldgaard) [reply apps/cookbook/src/app/component-status/component-status.component.scss:130]: I'm using this solution for now, let me know if you think it should be reverted.

**Comment by RasmusKjeldgaard** [comment]
I will have to look more closely into the failing test - I also have it locally, but initially thought it was related to my Mac (M1) like we have previously experienced. 

That shouldn't stop the review from happening though, I think we are very close.

**Comment by MadsBuchmann** [comment]
Should we add "closes #1365" to the PR description? We of course have to verify it does as you mention in [your own comment](https://github.com/kirbydesign/designsystem/issues/1365#issuecomment-1015377797).

**Comment by MadsBuchmann** [comment]
The base branch is `stable`. Remember to change it to something else before closing 😄

**Comment by RasmusKjeldgaard** [comment]
> The base branch is `stable`. Remember to change it to something else before closing 😄

Yep, this was just done for a more sane diff. I have converted to a draft now, and would like to make a new branch at a later point, where I redo the steps. Preferably without contaminating it all with a prettier update, which makes the diff huge.

**Comment by RasmusKjeldgaard** [comment]
Closed as it is replaced by #2140

**Pull request closed without merging**